plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControlImpl.java

```diff
@@ -0,0 +1,1005 @@
+            LOG.debug("<== getRowFilterResult(request=" + request + "): ret=" + ret);
+        }
+
+        return ret;
+    }
+
+    private boolean isDataMaskEnabled(RangerAccessResult result)
+    {
+        return result != null && result.isMaskEnabled();
+    }
+
+    private boolean isRowFilterEnabled(RangerAccessResult result)
+    {
+        return result != null && result.isRowFilterEnabled();
+    }
+
+    @Override
+    public List<ViewExpression> getRowFilters(SystemSecurityContext context, CatalogSchemaTableName tableName)
+    {
+        RangerTrinoAccessRequest request = createAccessRequest(createResource(tableName), context, TrinoAccessType.SELECT);
+        RangerAccessResult result = getRowFilterResult(request);
+
+        ViewExpression viewExpression = null;
+        if (isRowFilterEnabled(result)) {
+            String filter = result.getFilterExpr();
+
+            viewExpression = new ViewExpression(
+                    context.getIdentity().getUser(),
+                    Optional.of(tableName.getCatalogName()),
+                    Optional.of(tableName.getSchemaTableName().getSchemaName()),
+                    filter);
+        }
+
+        return List.of(viewExpression);
+    }
+
+    @Override
+    public List<ViewExpression> getColumnMasks(SystemSecurityContext context, CatalogSchemaTableName tableName, String columnName, Type type)
+    {
+        RangerTrinoAccessRequest request = createAccessRequest(
+                createResource(tableName.getCatalogName(), tableName.getSchemaTableName().getSchemaName(),
+                        tableName.getSchemaTableName().getTableName(), Optional.of(columnName)),
+                context, TrinoAccessType.SELECT);
+        RangerAccessResult result = getDataMaskResult(request);
+
+        ViewExpression viewExpression = null;
+        if (isDataMaskEnabled(result)) {
+            String maskType = result.getMaskType();
+            RangerServiceDef.RangerDataMaskTypeDef maskTypeDef = result.getMaskTypeDef();
+            String transformer = null;
+
+            if (maskTypeDef != null) {
+                transformer = maskTypeDef.getTransformer();
+            }
+
+            if (StringUtils.equalsIgnoreCase(maskType, RangerPolicy.MASK_TYPE_NULL)) {
+                transformer = "NULL";
+            }
+            else if (StringUtils.equalsIgnoreCase(maskType, RangerPolicy.MASK_TYPE_CUSTOM)) {
+                String maskedValue = result.getMaskedValue();
+
+                if (maskedValue == null) {
+                    transformer = "NULL";
+                }
+                else {
+                    transformer = maskedValue;
+                }
+            }
+
+            if (StringUtils.isNotEmpty(transformer)) {
+                transformer = transformer.replace("{col}", columnName).replace("{type}", type.getDisplayName());
+            }
+
+            viewExpression = new ViewExpression(
+                    context.getIdentity().getUser(),
```

Comment:
why current user? That means if you have recursive masks or filters then you may end up that you see filtered or masked data in subsequent expression.

Reply:
@dprophet

Reply:
In the FileBasedSystemAccessControl.java [getRowFilters](https://github.com/trinodb/trino/blob/master/lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java#L647)

They also context.getIdentity().getUser() in the same way I do recall.

DISCLAIMER (off topic): Nesting/recursive row level filters is not a good idea for performance reasons